src/index.tsx

```
@@ -1,100 +1,114 @@
-  TEN = '10',
-  HALF = 'half',
-  FULL = 'full',
+  TEN = "10",
+  HALF = "half",
+  FULL = "full",
 }
 
+app.post("/api/marathon-sequential", async (c) => {
+  const { firstName, lastName, email, address, distance } = await c.req.json();
 
-app.post('/api/marathon-sequential',async (c) => {
- 
-  const { firstName, lastName, email, address, distance } = await c.req.json()
-
-  await insertData(firstName, lastName, email, address, distance, c.env.DATABASE_URL);
-  await sendMail(email, c.env.RESEND_API, firstName)
-
-  return c.text('Thanks for registering for our Marathon', 200)
-
-})
-
-app.post('/api/marathon-fire-and-forget', async (c) => {
-  const { firstName, lastName, email, address, distance } = await c.req.json()
-  
-  await insertData(firstName, lastName, email, address, distance, c.env.DATABASE_URL).catch(console.error)
-  
-  sendMail(email, c.env.RESEND_API, firstName).catch(console.error)
-
-  
-    return c.text('Thanks for registering for our Marathon', 201)
+  await insertData(
+    firstName,
+    lastName,
+    email,
+    address,
+    distance,
+    c.env.DATABASE_URL,
+  );
+  await sendMail(email, c.env.RESEND_API, firstName);
 
-}
-)
-
-app.post('/api/marathon-waituntil', async (c) => {
+  return c.text("Thanks for registering for our Marathon", 200);
+});
 
-  const { firstName, lastName, email, address, distance } = await c.req.json()
+app.post("/api/marathon-fire-and-forget", async (c) => {
+  const { firstName, lastName, email, address, distance } = await c.req.json();
 
-  await insertData(firstName, lastName, email, address, distance, c.env.DATABASE_URL).catch(console.error)
-  c.executionCtx.waitUntil(sendMail(email, c.env.RESEND_API, firstName)); 
-  
-  return c.text('Thanks for registering for our Marathon', 201)
-  
+  await insertData(
+    firstName,
+    lastName,
+    email,
+    address,
+    distance,
+    c.env.DATABASE_URL,
+  );
 
-})
+  sendMail(email, c.env.RESEND_API, firstName).catch(console.error);
```

as a little javascript learning:

here it's fine to have the `.catch(console.error)` because whether or not we handle the exception does not affect the execution flow of the handler. 

why?

because we're not `await`'ing the call to `sendMail`.

when you `await` a promise, and the promise rejects (has an error), then you're telling javascript "hey i want to throw an error when something goes wrong in this promise"

throwing an error interrupts the flow of execution, so you'd never reach the code that returns a success message to the end user